src/api/web3/hooks.ts

```diff
@@ -0,0 +1,17 @@
+import { useMemo } from 'react'
```

Comment:
Are we also adding hooks in the API? 

I thought we were directly adding those to `hooks/`, maybe you are right and it makes sense and is coherent with what we do in the state. 

I think we should at some point review our current hooks and re-think about a structure for them.
Maybe some are even useful only in the context of v1, so they should be moved to `gp-v1` app probably

Reply:
Yeah, wasn't totally sure about this one either.
One the bright side, this is the approach 1, which I'm starting to like less in favour of approach 2.

In any case, we can always move this to hooks/ instead of leaving it in the API.